### PR TITLE
feat(vertexai): add name constructor for function calling schema

### PR DESCRIPTION
**Pull request by cynthiajoan**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart [hunk 1]
```diff
@@ -13,4 +13,4 @@
 // limitations under the License.
 
 /// version number for the package, should be align with pubspec.yaml.
-const packageVersion = '0.1.0';
+const packageVersion = '0.2.0';
```
Comment (natebosch) [inline packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart:16]: Nothing in this PR looks breaking to me - we could bump to `0.1.1` if we don't have any other breaking changes.

**Review by natebosch — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]
@cynthiajoan - We've just merged a PR that will automatically update the `version_version.dart` when we publish new vertexai versions. So there's a merge conflict I'm afraid 😓. Other than that, LGTM.

File: packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart (hunk 1 above)
Comment (cynthiajoan) [reply packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_version.dart]: I think the 0.2.0 is introduced when the whole flutterfire has breaking change. This file should be updated automatically in the future for release. Currently resolved to use the existing file in the default branch

**Comment by cynthiajoan** [comment]
> @cynthiajoan - We've just merged a PR that will automatically update the `version_version.dart` when we publish new vertexai versions. So there's a merge conflict I'm afraid 😓. Also analyse and format CI needs fixing. Other than that, LGTM.

resolved the version

**Pull request merged**